### PR TITLE
Adding I2C documentation for Raspberry Pi

### PR DESCRIPTION
**Pull request by Ellerbach**
- Adding I2C documentation for Raspberry Pi
-One more from the #802 list

### CODE REVIEW

File: Documentation/raspi-i2c.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+Save the file with `ctrl + x` then `Y` then `enter`
+
+Then reboot:
+
+```bash
+sudo reboot
+```
+
+More information on dtoverlay and how to select specific elements of I2C buses are [available here](https://github.com/raspberrypi/firmware/blob/bff705fffe59ad3eea33999beb29c3f26408de40/boot/overlays/README#L1387). 2 busses are available on any Raspberry Pi. More can be available on specific models like those base out of BCM2711.
```
Comment (pgrawehr) [inline Documentation/raspi-i2c.md:69]: I suggest adding "such as Raspberry Pi 4" somewhere. That's what people know better than the CPU it runs with.

File: Documentation/raspi-i2c.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+You should find a line like this one:
+
+```text
+SUBSYSTEM=="i2c-dev", GROUP="i2c", MODE="0660"
+```
+
+If you don't have it or if you want to adjust the permissions, this is what you'll need to add/adjust, as always save through `ctrl + x` then `Y` then `enter` and then reboot.
+
+## Bonus: write your own I2C scanner
```
Comment (pgrawehr) [inline Documentation/raspi-i2c.md:129]: Maybe add a hint that there's already an existing tool to do this?

File: Documentation/raspi-i2c.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+```text
+Unhandled exception. System.IO.IOException: Error 121 performing I2C data transfer.
+   at System.Device.I2c.UnixI2cDevice.ReadWriteInterfaceTransfer(Byte* writeBuffer, Byte* readBuffer, Int32 writeBufferLength, Int32 readBufferLength)
+   at System.Device.I2c.UnixI2cDevice.Transfer(Byte* writeBuffer, Byte* readBuffer, Int32 writeBufferLength, Int32 readBufferLength)
+   at System.Device.I2c.UnixI2cDevice.WriteByte(Byte value)
+   at TestTest.Program.Main(String[] args) in C:\tmp\TestI2C\Program.cs:line 14
+Aborted
+```
+
```
Comment (pgrawehr) [inline Documentation/raspi-i2c.md:42]: Add: "In rare cases, you might see the above exception during normal operation of a device as well. Adding retries might solve the issue."

**Review by pgrawehr — APPROVED** [review]

File: Documentation/raspi-i2c.md (hunk 3 above)
Comment (Ellerbach) [reply Documentation/raspi-i2c.md:42]: Yes, good point. It does not exist on all systems. But right, it does on some! Will add!

File: Documentation/raspi-i2c.md (hunk 2 above)
Comment (Ellerbach) [reply Documentation/raspi-i2c.md:129]: Yes, good point. It does not exist on all systems. But right, it does on some! Will add!

File: Documentation/raspi-i2c.md (hunk 1 above)
Comment (Ellerbach) [reply Documentation/raspi-i2c.md:69]: ok!

File: Documentation/raspi-i2c.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+```bash
+sudo reboot
+```
+
+More information on dtoverlay and how to select specific elements of I2C buses are [available here](https://github.com/raspberrypi/firmware/blob/bff705fffe59ad3eea33999beb29c3f26408de40/boot/overlays/README#L1387). 2 busses are available on any Raspberry Pi. More can be available on specific models like those base out of BCM2711 such as Raspberry Pi 4.
+
+### Enabling I2C with advance parameters
+
+The general pattern looks like the following:
```
Comment (krwq) [inline Documentation/raspi-i2c.md:75]: you should also mention you can do `sudo raspi-config nonint do_i2c 0` on raspbian

**Review by krwq — APPROVED** [review]

File: Documentation/raspi-i2c.md (hunk 4 above)
Comment (Ellerbach) [reply Documentation/raspi-i2c.md:77]: Added. 1 is the default, so changed for 1 with a note

**Review by krwq — APPROVED** [review]

File: Documentation/raspi-i2c.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+Then reboot:
+
+```bash
+sudo reboot
+```
+
+More information on dtoverlay and how to select specific elements of I2C buses are [available here](https://github.com/raspberrypi/firmware/blob/bff705fffe59ad3eea33999beb29c3f26408de40/boot/overlays/README#L1387). 2 busses are available on any Raspberry Pi. More can be available on specific models like those base out of BCM2711.
+
+### Enabling I2C with advance parameters
```
[the remark this answers is not in the document]
Comment (Ellerbach) [reply Documentation/raspi-i2c.md:75]: I have to say I prefer all in one. Makes it easier. The first part may solve 99% of the problems, knowing you have all in 1 page makes you feel that you'll find a solution whatever happens.

[the remark this answers is not in the document]
Comment (krwq) [reply Documentation/raspi-i2c.md:75]: Yes I agree with @Ellerbach, we can maybe restructure doc a bit to cover 99% cases in the first paragraph but I think it's not really that large. What I think we should do next is to add a category in each device using I2cDevice/SpiDevice (perhaps even automate that) and add links to these instructions in each sample so people can easily find it

**Pull request merged**